packages/data-table-server/src/routes/FetchPreviewData.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+
+import { Route } from '@tupaia/server-boilerplate';
+import { DataTableType } from '../models';
+
+export type FetchPreviewDataRequest = Request<
+  Record<string, unknown>,
+  { rows: unknown[]; columns: unknown[] },
+  DataTableType,
+  Record<string, unknown>
+>;
+
+export class FetchPreviewDataRoute extends Route<FetchPreviewDataRequest> {
+  public async buildResponse() {
+    const { body, ctx } = this.req;
+    const LIMIT = 200;
+
+    const requestParams = { ...body };
+    const data = await ctx.dataTableService.fetchData(requestParams);
+    let columnArray = new Set();
+    for (const row of data) {
+      if (typeof row === 'object' && row) {
+        columnArray = new Set([...columnArray, ...Object.keys(row).map(key => key)]);
+      }
+    }
+
+    return {
+      rows: data.splice(0, LIMIT),
+      columns: Array.from(columnArray),
+      total: data.length,
+      limit: LIMIT,
```

Comment:
It is devastating to pass options to `knex` to add limit and pagination. 

I think it should be fine to do it here. It could take ~30 secs to fetch all records though.

Reply:
Oh yep so the idea was to prevent the 30 sec wait. The limit on the way back after the SQL query has completed doesn't do much.

Can we implement this in SqlDataTableService by wrapping the sql query:
```
const limit = 1000;
const wrappedResultQuery = 'SELECT * FROM ${inputSql} LIMIT ${limit}';
const wrappedCountQuery = 'SELECT count(*) FROM ${inputSql}';
``` 

The Postgres Query Plan Builder should be smart enough to optimise such these queries to be super quick.

Reply:
Great example, thanks a lot!